### PR TITLE
Angular codegen

### PR DESCRIPTION
**Pull request by santam85**

### CODE REVIEW

File: boat-engine/src/main/java/com/backbase/oss/boat/transformers/bundler/BoatDeserializationUtils.java [hunk 1]
```diff
@@ -39,11 +39,11 @@
     }
 
     private static boolean isJson(String contents) {
-        return contents.toString().trim().startsWith("{");
+        return contents.trim().startsWith("{");
```
Comment (bartveenstra) [inline boat-engine/src/main/java/com/backbase/oss/boat/transformers/bundler/BoatDeserializationUtils.java:42]: NullpointerException waiting to happen. Maybe check if it's null?

File: boat-scaffold/src/main/java/com/backbase/oss/codegen/angular/BoatAngularGenerator.java [hunk 2]
```diff
@@ -0,0 +1,831 @@
...
+            additionalProperties.put("zonejsVersion", "0.9.1");
+        } else {
+            // compatible versions to Angular 6+
+            additionalProperties.put("zonejsVersion", "0.8.26");
+        }
+
+        //Files for building our lib
+        supportingFiles.add(new SupportingFile("package.mustache", getIndexDirectory(), "package.json"));
+        supportingFiles.add(new SupportingFile("tsconfig.mustache", getIndexDirectory(), "tsconfig.json"));
+    }
```
Comment (bartveenstra) [inline boat-scaffold/src/main/java/com/backbase/oss/codegen/angular/BoatAngularGenerator.java:350]: This entire section seems quite long. Can't we call super on this one?

File: boat-scaffold/src/main/templates/boat-angular-mock/licenseInfo.mustache [hunk 3]
```diff
@@ -0,0 +1,11 @@
+/**
+ * {{{appName}}}
+ * {{{appDescription}}}
+ *
+ * {{#version}}The version of the OpenAPI document: {{{version}}}{{/version}}
+ * {{#infoEmail}}Contact: {{{infoEmail}}}{{/infoEmail}}
+ *
+ * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
```
Comment (bartveenstra) [inline boat-scaffold/src/main/templates/boat-angular-mock/licenseInfo.mustache:8]: Wel....Is it? Or should we use boat here ;-)

**Review by bartveenstra — CHANGES_REQUESTED** [review]

File: boat-scaffold/src/main/java/com/backbase/oss/codegen/angular/BoatAngularGenerator.java (hunk 2 above)
Comment (santam85) [reply boat-scaffold/src/main/java/com/backbase/oss/codegen/angular/BoatAngularGenerator.java:350]: Those configs are not present on the Abstract Typescript one, I believe...

**Review by bartveenstra — APPROVED** [review]

**Comment by bartveenstra** [comment]
Good stuff. I am merging coverage first to dev, and then i'll merge dev into this -> test -> merge

**Review by bartveenstra — APPROVED** [review]

**Comment by santam85** [comment]
@bartveenstra let's wait for the last changes from @tekerson , then we can go for it

**Pull request merged**